docs/source/getting_started/getting_started_hard_way.rst

```diff
@@ -0,0 +1,269 @@
+
+.. note::  Docker environment will be removed when you detach from the container.
+
+Linux
+^^^^^
+
+Boost
+"""""
+
+To install Boost libraries (``libboost-all-dev``), use `current release <http://www.boost.org/users/download/>`_ from Boost webpage, or use `debian repository <https://packages.debian.org/sid/libboost-all-dev>`_ for latest library. The only dependencies are system and filesystem, so use ``./bootstrap.sh --with-libraries=system,filesystem`` when you are building the project.
+
+Other dependencies
+""""""""""""""""""
+
+Use this code to install dependencies on Debian-based Linux distro.
+
+.. code-block:: shell
+
+  apt-get update; \
+  apt-get -y --no-install-recommends install \
+  build-essential automake libtool \
+  libssl-dev zlib1g-dev \
+  libc6-dbg golang \
+  git ssh tar gzip ca-certificates \
+  wget curl file unzip \
+  python
+
+.. note::  If you are willing to actively develop Iroha, please consider installing the `latest release <https://cmake.org/download/>`_ of CMake.
+
+macOS
+^^^^^
+If you are a happy macOS user and are willing to play with Iroha without building it, you can install our official Homebrew package and go straight to `Playing around with settings` section
+
+.. code-block:: shell
+
+  brew tap soramitsu/iroha
+  brew install iroha
+
+If you want to build it from scratch and actively develop it, please use this code to install all dependencies with Homebrew.
+
+.. code-block:: shell
+
+  xcode-select --install
+  brew install cmake boost postgres grpc autoconf automake libtool golang libpqxx
+
+.. hint:: To install the Homebrew itself please run 
+
+  ``ruby -e "$(curl -fsSL https://raw.githubusercontent.com/homebrew/install/master/install)"``
+
+Getting Iroha build
+-------------------
+
+Cloning the repository
+^^^^^^^^^^^^^^^^^^^^^^
+Clone the `Iroha repository <https://github.com/hyperledger/iroha>`_ to the directory of your choice.
+
+.. code-block:: shell
+
+  git clone -b develop https://github.com/hyperledger/iroha
+  cd iroha
+
+.. hint:: If you have installed the prerequisites with Docker, you don't need to clone Iroha again
+
+
+Building Iroha
+^^^^^^^^^^^^^^
+To build Iroha, use those commands
+
+.. code-block:: shell
+
+  cmake -H. -Bbuild;
+  cmake --build build -- -j$(nproc)
+
+.. note::  On macOS ``$(nproc)`` variable does not work. Check number of logical cores with ``sysctl -n hw.ncpu`` and put it explicitly in the command above, e.g. ``cmake --build build -- -j4``
+
+CMake parameters
+^^^^^^^^^^^^^^^^
+
+We use CMake to build platform-dependent build files. It has numerous of the flags for configuring the final build. Note that beside the listed params cmake's vars can be useful as well. Also as long as this page can be deprecated (or just not complete) you can browse custom flags via ``cmake -L``, ``cmake-gui``, or ``ccmake``.
```

Comment:
https://www.englishforums.com/English/NumerousOfThe/qqmll/post.htm

has numerous flags

Reply:
besides the listed

https://dictionary.cambridge.org/grammar/british-grammar/beside-or-besides

Reply:
params — parameters
vars — variables